ClosedXML/Excel/AutoFilters/XLFilterColumn.cs

```diff
@@ -233,7 +233,7 @@ private void SetTopBottom(Int32 value, XLTopBottomType type, Boolean takeTop = t
                                        .SetTopBottomValue(value)
                                        .SetTopBottomType(type);
 
-            var values = GetValues(value, type, takeTop);
+            var values = GetValues(value, type, takeTop).ToArray();
```

Comment:
`.ToArray()` obviously has some overhead, both in terms of CPU cycles and memory. So unless it reduces enumerations, I would like to avoid it. From what I can see, this enumerable is evaluated only once in line 248's `foreach (double val in values)`. So how would this `.ToArray()` improve performance?

Reply:
The loop you are referring to is inside the other loop so this change makes sense.

Reply:
@Pankraty has explained the idea.
In additional I would say: IEnumerable is a great thing to implement lazy computation, but then and only then we properly materialize encapsulated value.